libutils/sequence.h

```diff
@@ -115,8 +115,20 @@ typedef int (*SeqItemComparator) (const void *, const void *, void *user_data);
  */
 int StrCmpWrapper(const void *s1, const void *s2, void *user_data);
 
+/**
+ * Replace value at #index.
+ *
+ * @warning Destroys the original item at #index! See SeqSoftSet().
+ */
 void SeqSet(Seq *set, size_t index, void *item);
 
+/**
+ * Set value at #index.
+ *
+ * @note Make sure the original item at #index is destroyed.
```

Comment:
Adding to my comment above, perhaps we could make sure that we _cant_ discard the return value with a compiler attribute like `FUNC_WARN_UNUSED_RESULT`?
That way we at least know that the compiler will warn us about potential leak.

Reply:
Yup, for that new function. :wink: